src/Common/src/Interop/Unix/Interop.IOErrors.cs

```diff
@@ -17,49 +17,45 @@ private static bool CheckIo(ErrorInfo errorInfo, string path, bool isDirectory,
             errorInfo = errorRewriter(errorInfo);
         }
 
-        if (errorInfo.Error != Error.EINTR)
-        {
-            throw Interop.GetExceptionForIoErrno(errorInfo, path, isDirectory);
-        }
-
-        return true;
+        throw Interop.GetExceptionForIoErrno(errorInfo, path, isDirectory);
```

Comment:
Consider adding an assert that errorInfo is not EINTR.  If that assert fires, it means we missed something in the native layer that needs to check for it.